File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02574/p02574_num1_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

// Function to calculate the greatest common divisor (GCD) of two numbers
int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
}

int main() {
    int N;
    cin >> N;

    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    int gcdAll = A[0];
    int gcdPairwise = A[0];
    for (int i = 1; i < N; i++) {
        gcdAll = gcd(gcdAll, A[i]);
        gcdPairwise = gcd(gcdPairwise, A[i]);
    }

    if (gcdAll != 1) {
        cout << "not coprime" << endl;
    } else if (gcdPairwise == 1) {
        cout << "pairwise coprime" << endl;
    } else {
        cout << "setwise coprime" << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02574/p02574_num1_parsed.py
'''
import math

# function to check if two numbers are coprime
def is_coprime(a, b):
    return math.gcd(a, b) == 1

# function to check if a list of numbers is pairwise coprime
def is_pairwise_coprime(numbers):
    n = len(numbers)
    for i in range(n):
        for j in range(i+1, n):
            if not is_coprime(numbers[i], numbers[j]):
                return False
    return True

# function to check if a list of numbers is setwise coprime
def is_setwise_coprime(numbers):
    gcd = numbers[0]
    for i in range(1, len(numbers)):
        gcd = math.gcd(gcd, numbers[i])
    return gcd == 1

# read input
N = int(input())
A = list(map(int, input().split()))

# check if the numbers are pairwise coprime
if is_pairwise_coprime(A):
    print("pairwise coprime")
elif is_setwise_coprime(A):
    print("setwise coprime")
else:
    print("not coprime")
'''